### PR TITLE
OOP school library: add Person, Student and Teacher classes

### PR DESCRIPTION
**Pull request by akos29**
### OOP school library: add Person, Student, and Teacher classes

In this request majorly:

- [ ] add class `Person` with  `constructor`, `getters and setters `, `private`, and `public` methods 
- [ ] add `Student` class that inherits from the `Person class's
- [ ] add `Teacher` class with specialization

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+require './person'
+
+class Student < Person
+  def initialize(name, classroom = 'unknown')
+    super(name)
```
Comment (Olamarx) [inline student.rb]: - [ ] Constructor extends parent's constructor by adding @classroom and a parameter for it and the duper needs to access the three parameters i.e `super(age, name, parent_permission)`. You can learn more [here](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/class_and_object_for_irb.md#constructor-and-instance-vars)

N.B: same for the teacher's class too.

**Review by Olamarx — CHANGES_REQUESTED** [review]
Hi @akos29,

Good job so far, indeed you have spent time on this project!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Linters passed ✔️
- Commit messages ✔️ 

## Required Changes ♻️
_Check the comments under the review._
- [x] Kindly fix the MIT license link, it currently doesn't direct to the appropriate page. This page should be the permission page to use this repo.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by prolajumokeoni** [comment]
Hi @akos29, kindly fix the linter errors 👍🏿

**Comment by akos29** [comment]
> Hi @akos29, kindly fix the linter errors 👍🏿

Hi @prolajumokeoni, I just fixed the errors. I kindly ask you to check my work 😊

**Review by prolajumokeoni — APPROVED** [review]
Hi @akos29 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
- No linter errors ✅ 
- Git flow was used correctly 🚀 
- Good Pr title and description 💪🏿 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**